crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -28,25 +28,35 @@ use crate::{
 /// This is done by setting the values of their [`ComputedVisibility`] component.
 #[derive(Component, Clone, Reflect, Debug)]
 #[reflect(Component, Default)]
-pub struct Visibility {
-    /// Indicates whether this entity is visible. Hidden values will propagate down the entity hierarchy.
-    /// If this entity is hidden, all of its descendants will be hidden as well. See [`Children`] and [`Parent`] for
-    /// hierarchy info.
-    pub is_visible: bool,
+pub enum Visibility {
+    Visible,
+    Invisible,
 }
 
 impl Default for Visibility {
     fn default() -> Self {
-        Visibility::VISIBLE
+        Self::Visible
     }
 }
 
 impl Visibility {
-    /// A [`Visibility`], set as visible.
-    pub const VISIBLE: Self = Visibility { is_visible: true };
+    /// Whether this entity is visible.
+    #[inline]
+    pub const fn is_visible(&self) -> bool {
+        match self {
+            Self::Visible => true,
+            Self::Invisible => false,
+        }
+    }
```

Comment:
I suggest to use `Eq` instead of this kind of utility method for field-less enum.

Reply:
Agreed.